Mathlib/Data/List/Sigma.lean

```diff
@@ -282,6 +282,24 @@ theorem perm_lookupAll (a : α) {l₁ l₂ : List (Sigma β)} (nd₁ : l₁.Nodu
     (p : l₁ ~ l₂) : lookupAll a l₁ = lookupAll a l₂ := by
   simp [lookupAll_eq_dlookup, nd₁, nd₂, perm_dlookup a nd₁ nd₂ p]
 
+theorem dlookup_append_of_not_mem_left {l₀ : List (Sigma β)} {a : α} (h : a ∉ l₀.keys)
+    (l₁ : List (Sigma β)) : (l₀ ++ l₁).dlookup a = l₁.dlookup a := by
```

Comment:
Is this true more generally as
```suggestion
theorem dlookup_append_of_not_mem_left {l₀ : List (Sigma β)} {a : α}
    (l₁ : List (Sigma β)) : (l₀ ++ l₁).dlookup a = l₀.dlookup a <|> l₁.dlookup a := by
```

Reply:
Yeah! That's a much nicer way to state it.